utils/chunked_vector.hh

```diff
@@ -351,7 +351,8 @@ chunked_vector<T, max_contiguous_allocation>::chunked_vector(Iterator begin, Ite
 }
 
 template <typename T, size_t max_contiguous_allocation>
-chunked_vector<T, max_contiguous_allocation>::chunked_vector(size_t n, const T& value) {
+chunked_vector<T, max_contiguous_allocation>::chunked_vector(size_t n, const T& value)
+        : chunked_vector() {
```

Comment:
Is this enough to guarantee exception safety?

Reply:
No, as the unit test that wasn't added proves.

(I'm guessing)

Reply:
It is enough for calling the destructor if the body throws an exception, see https://godbolt.org/z/85W1aEjfE
And since the back_inserter updates `_size` for each insertion, the destructor will destroy all the items that were filled successfully.

In the context of https://github.com/scylladb/scylladb/pull/18609, we can optimize this and fill one chunk at a time using std::uninitialized_fill_n, that is exception safe.
See https://en.cppreference.com/w/cpp/memory/uninitialized_fill_n
```
If an exception is thrown during the initialization, the objects already constructed are destroyed in an unspecified order.
```

I can add unit tests to prove that :)

Reply:
unit test added in v2